content/Documentation Guidelines - Sebastián Caraballo.md

```diff
@@ -0,0 +1,112 @@
+## Why is Documentation Important For Development?
+
+![main image](images/why_documenting_matters.jpg)
+
+
+Here at Rootstrap, we help our clients reach their business goals by developing high-quality, performant, and scalable software solutions. To achieve this, our team continuously focuses on improving all aspects of our processes, through several initiatives across all company departments.
+
+In this article, we will examine why documentation is vital for the software development cycle.
+
+### Why is Documentation so Vital?
+
+The reader may be wondering about the benefits of documenting. Let me tell you a brief story. In March 2020, we started working with a client on a very exciting project. However, the whole team switched to remote work due to COVID-19. So we had to focus on how to keep a fluent communication among frontend & backend developers. In order to do this, the backend team started to document the API features and the available deliverables right away with different tools. Doing this helped to maintain the team's velocity, reduce synchronization time between devs, and keep everyone aware of each other's work.
+
+Many of the features require backend work to be done first so that frontenders can integrate their mocked code with the API afterwards. This now implies that backenders not only have to code and test, but also add documentation on each new released feature. By the time a frontender starts working with any API endpoint, they already have a detailed description of returned objects, whether authentication is needed or not, and error messages that may arise based on different use case scenarios.
```

Comment:
```suggestion
Many features require backend work to be done first so that frontend devs can integrate their mocked code with the API afterward. This now implies that backend devs not only have to code and test but also add documentation on each newly released feature. By the time a frontend dev starts working with any API endpoint, they already have a detailed description of returned objects. They will also be aware of whether authentication is needed or not and will see any error messages that may arise based on different use case scenarios.
```